### PR TITLE
New Ad4Game adapter for Prebid.js 1.0

### PR DESCRIPTION
**Pull request by adilets**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ x] New bidder adapter   https://github.com/prebid/prebid.github.io/pull/333/files  
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change
Adding new bidder adapter for A4G

- test parameters for validating bids
```
{
  bidder: 'a4g',
  params: {
    zoneId: 59304,
    deliveryUrl: 'http://dev01.ad4game.com/v1/bid'
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer: adilet@ad4game.com, jaafar@ad4game.com, tech@ad4game.com
- [ ] official adapter submission

### CODE REVIEW

**Comment by adilets** [comment]
Hi @matthewlane. I created new PR from (https://github.com/prebid/Prebid.js/pull/1650). 

About response, you should use real domain instead of localhost when you testing. In my case I use `xmmorpg.com` domain. I rewrite localhost to this domain. 

Thank you

**Comment by dbemiller** [comment]
> About response, you should use real domain instead of localhost when you testing. In my case I use xmmorpg.com domain. I rewrite localhost to this domain.

hey @adilets, could you add a note about this in your `.md` file? It's nice to have more permanent documentation about caveats like this when we re-test 6 months from now.

File: modules/a4gBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      zoneIds.push(bid.params.zoneId);
+    });
+
+    if (!deliveryUrl) {
+      deliveryUrl = A4G_DEFAULT_BID_URL;
+    }
+
+    return {
+      method: 'GET',
+      url: deliveryUrl,
```
Comment (matthewlane) [inline modules/a4gBidAdapter.js]: One way to do it is to add `bidId` to this object, and set it's value to the bid id of `validBidRequests` (`bidId: validBidRequests[0] && validBidRequests[0].bidId` in that case). Then `bidId` will be available on the `request` parameter in `interpretResponse` and line 61 will work without modification. You can do it however you'd prefer though

File: modules/a4gBidAdapter.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                    bidder: 'a4g',
+                    params: {
+                        zoneId: 59354,
+                        deliveryUrl: 'http://dev01.ad4game.com/v1/bid'
+                    }
+               }
+           ]
+       }
+    ];
+```
```
Comment (matthewlane) [inline modules/a4gBidAdapter.md]: Please add a note about rewriting localhost to the test domain in order to receive bids

**Comment by adilets** [comment]
@matthewlane Thanks for your review and comments. I updated PR.

**Pull request merged**